plan/expression_rewriter.go

```diff
@@ -80,7 +80,11 @@ func (b *planBuilder) rewrite(expr ast.ExprNode, p LogicalPlan, aggMapper map[*a
 	if getRowLen(er.ctxStack[0]) != 1 {
 		return nil, nil, ErrOperandColumns.GenByArgs(1)
 	}
-	result := expression.FoldConstant(er.ctxStack[0])
+	result, err := expression.ConvertConstantType(er.ctxStack[0])
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	result = expression.FoldConstant(result)
```

Comment:
Well, you are right. Otherwise we cannot make query like `select * from t where dt > 20030914+1` return right results.